lib/promscrape/discovery/kuma/api.go

```diff
@@ -0,0 +1,207 @@
+package kuma
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
+	"github.com/VictoriaMetrics/metrics"
+)
+
+var configMap = discoveryutils.NewConfigMap()
+
+type apiConfig struct {
+	client *discoveryutils.Client
+	path   string
+
+	cancel        context.CancelFunc
+	wg            sync.WaitGroup
+	mu            sync.Mutex // protects targets
+	targets       []kumaTarget
+	latestVersion string
+	latestNonce   string
+
+	fetchErrors *metrics.Counter
+	parseErrors *metrics.Counter
+}
+
+const (
+	discoveryNode      = "victoria-metrics"
+	xdsApiVersion      = "v3"
+	xdsRequestType     = "discovery"
+	xdsResourceType    = "monitoringassignments"
+	xdsResourceTypeUrl = "type.googleapis.com/kuma.observability.v1.MonitoringAssignment"
+)
+
+var waitTime = flag.Duration("promscrape.kuma.waitTime", 0, "Wait time used by Kuma service discovery. Default value is used if not set")
```

Comment:
This description is copy-pasted from consul and nomad service discovery - see d4aadba9fa1921ba824e4079da205fa74048083c and https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1144 . Probably it can be updated there to proper value if needed.

BTW, the 9fbd45a22fdba49fd496aa6d2760b1cd4e44a46f removes the `-promscrape.kuma.waitTime` command-line flag, since it is unclear which benefits does the long polling give for Kuma service discovery:
- Kuma server always returns the full response. It cannot return updates to the state like Kubernetes API server does.
- Kuma server returns `304 Not Modified` response when there are no modifications in the state since the previous request.

Due to these reasons the commit 9fbd45a22fdba49fd496aa6d2760b1cd4e44a46f switches to blocking service discovery in order to simplify the code.